metpy/calc/tools.py

```diff
@@ -286,47 +287,135 @@ def reduce_point_density(points, radius, priority=None):
 
 
 @exporter.export
-def log_interp(x, xp, fp, **kwargs):
-    r"""Interpolates data with logarithmic x-scale.
+def log_interp(x, xp, *args, **kwargs):
+    r"""Interpolates data with logarithmic x-scale over a specified axis.
 
     Interpolation on a logarithmic x-scale for interpolation values in pressure coordintates.
 
     Parameters
     ----------
     x : array-like
-        The x-coordinates of the interpolated values.
+        1-D array of desired interpolated values.
 
     xp : array-like
         The x-coordinates of the data points.
 
-    fp : array-like
-        The y-coordinates of the data points, same length as xp.
+    args : array-like
+        The y-coordinates of the data points, same shape as xp.
+
+    axis : int, optional
+        The axis to interpolate over. Defaults to 0.
+
+    extrap: string, optional
```

Comment:
What about making this `fill_value` which defines the value to fill out of bounds points with? Could use `None` to raise a `ValueError` and could default to `np.nan`.